lighthouse-core/audits/byte-efficiency/unused-javascript.js

```diff
@@ -0,0 +1,184 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const Audit = require('./byte-efficiency-audit');
+const URL = require('../../lib/url-shim');
+
+const IGNORE_THRESHOLD_IN_BYTES = 2048;
+
+class UnusedJavaScript extends Audit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'Performance',
+      name: 'unused-javascript',
+      description: 'Unused JavaScript',
+      informative: true,
+      helpText: 'Remove unused JavaScript to reduce bytes consumed by network activity.',
+      requiredArtifacts: ['JsUsage', 'networkRecords']
+    };
+  }
+
+  /**
+   * @param {{used: number, unused: number}} objA
+   * @param {{used: number, unused: number}} objB
+   * @return {{used: number, unused: number}}
+   */
+  static _add(objA, objB) {
```

Comment:
`_mergeRanges`? or something more specific like that. And `rangeA`, `rangeB` would be better param names, even if they are sometimes ranges and sometimes just `{used: number, unused: number}`